lib/dal/src/workspace_snapshot/node_weight/secret_node_weight.rs

```diff
@@ -0,0 +1,198 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use si_events::{merkle_tree_hash::MerkleTreeHash, ulid::Ulid, ContentHash, EncryptedSecretKey};
+
+use crate::workspace_snapshot::content_address::ContentAddressDiscriminants;
+use crate::workspace_snapshot::vector_clock::VectorClockId;
+use crate::EdgeWeightKindDiscriminants;
+use crate::{
+    change_set::ChangeSet,
+    workspace_snapshot::{
+        content_address::ContentAddress,
+        graph::LineageId,
+        node_weight::{NodeWeightError, NodeWeightResult},
+        vector_clock::VectorClock,
+    },
+};
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct SecretNodeWeight {
+    id: Ulid,
+    lineage_id: LineageId,
+    content_address: ContentAddress,
+    merkle_tree_hash: MerkleTreeHash,
+    vector_clock_first_seen: VectorClock,
+    vector_clock_recently_seen: VectorClock,
+    vector_clock_write: VectorClock,
+    encrypted_secret_key: EncryptedSecretKey,
```

Comment:
The purpose of the node weight is to put the key onto the node. We will need to have quick access to all secrets by key, so this ensures we can remain entirely in memory.